### PR TITLE
[SPARK-48300][SQL] Codegen Support for `from_xml`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The PR aims to add `Codegen Support` for `from_xml`

### Why are the changes needed?
- Improve codegen coverage.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
- Add new UT & existed UT.
- Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/xmlExpressions.scala [hunk 1]
```diff
@@ -104,7 +103,7 @@ case class XmlToStructs(
     if (mode != PermissiveMode && mode != FailFastMode) {
       throw QueryCompilationErrors.parseModeUnsupportedError("from_xml", mode)
     }
-    val (parserSchema, actualSchema) = nullableSchema match {
```
Comment (panbingkun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/xmlExpressions.scala:107]: The variable `actualSchema` is not used anywhere

**Comment by panbingkun** [comment]
cc @sandip-db @HyukjinKwon @cloud-fan @yaooqinn

**Comment by sandip-db** [comment]
@panbingkun Thanks for submitting the PR. Can you please separate the `codegen` support and the cleanup in separate PRs?

**Comment by panbingkun** [comment]
> @panbingkun Thanks for submitting the PR. Can you please separate the `codegen` support and the cleanup in separate PRs?
Sure,
A new separate PR for `cleanup`: https://github.com/apache/spark/pull/46674

**Comment by panbingkun** [comment]
After the PR above is merged, I will rebase the PR again just for `codegen`
Thanks.

**Comment by panbingkun** [comment]
It has rebase the master. At present, this PR is only for `codegen support for from_xml`
@sandip-db @HyukjinKwon @yaooqinn @cloud-fan

File: sql/core/src/test/scala/org/apache/spark/sql/XmlFunctionsSuite.scala [hunk 2]
```diff
@@ -40,6 +40,17 @@ class XmlFunctionsSuite extends QueryTest with SharedSparkSession {
       Row(Row(1)) :: Nil)
   }
 
+  test("SPARK-48300: from_xml - Codegen Support") {
+    withTempView("XmlToStructsTable") {
+      val dataDF = Seq("""<ROW><a>1</a></ROW>""").toDF("value")
+      dataDF.createOrReplaceTempView("XmlToStructsTable")
+      val df = sql("SELECT from_xml(value, 'a INT') FROM XmlToStructsTable")
+      val plan = df.queryExecution.executedPlan
+      assert(plan.isInstanceOf[WholeStageCodegenExec])
```
Comment (yaooqinn) [inline sql/core/src/test/scala/org/apache/spark/sql/XmlFunctionsSuite.scala:49]:
```suggestion
      assert(df.queryExecution.executedPlan.isInstanceOf[WholeStageCodegenExec])
```

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
Merged to master.

Thank you @panbingkun

**Pull request closed without merging**